wallet_migrator/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+pub mod credx2anoncreds;
+pub mod error;
+pub mod vdrtools2credx;
+
+use std::fmt::Display;
+
+use error::MigrationResult;
+use log::{debug, error, info, warn};
+pub use vdrtools::{
+    types::domain::wallet::{Config, Credentials, KeyDerivationMethod, Record},
+    Locator, WalletHandle,
+};
+
+/// Retrieves all records from a wallet and migrates them
+/// by applying the `migrate_fn` argument.
+///
+/// The migrated records are inserted into a newly created
+/// wallet, based on the provided `config` and `credentials`.
+pub async fn migrate_wallet<E>(
+    wallet_handle: WalletHandle,
+    config: Config,
+    credentials: Credentials,
+    migrate_fn: impl FnMut(Record) -> Result<Option<Record>, E>,
+) -> MigrationResult<()>
+where
+    E: Display,
+{
+    info!("Starting wallet migration...");
+    let locator = Locator::instance();
+
+    debug!("Creating new wallet {} ...", config.id);
+    locator
+        .wallet_controller
+        .create(config.clone(), credentials.clone())
+        .await?;
+
+    debug!("Opening new wallet...");
+
+    let new_wh = locator
+        .wallet_controller
+        .open(config.clone(), credentials.clone())
+        .await?;
+
+    debug!("Migrating records from wallet with handle {wallet_handle:?} to wallet with handle {new_wh:?}");
+
+    let res = locator
+        .wallet_controller
+        .migrate_records(wallet_handle, new_wh, migrate_fn)
```

Comment:
Would be nice to have dry run option. If I run this on server with many records, it'd be nice to first just check the migration can succeed given the current wallet state, avoiding any risks/possible issue related to writing to db.

Reply:
It would be cool if we could also count number of records as part of the dry run (or perhaps regardless)

Reply:
Hmmmmmm... It would be a bit tricky to implement as part of the library because it would have to mess up with all the internals of the migration code.

Would an "ephemeral" SQLite wallet do the trick? I'd personally just migrate (and create a new wallet) and ensure to delete it after (maybe use some UUID as part of the name for uniqueness and to avoid concurrency issues).

I could add some helper function for this but idk, it feels like something too small and specific to be worth it. What do you think?

Reply:
Just to be more specific, the helper function would literally just do something like:
```rust
async fn dry_run( 
    wallet_handle: WalletHandle,
    config: Config,
    credentials: Credentials,
    migrate_fn: impl FnMut(Record) -> Result<Option<Record>, E>,
) -> MigrationResult<()>
where
    E: Display,
{
    if migrate_wallet(wallet_handle, config, credentials, migrate_fn).await.is_ok() {
        let new_wh = Locator::instance()
            .wallet_controller
            .open(config.clone(), credentials.clone())
            .await
            .unwrap();

        Locator::instance().wallet_controller.close(new_wh).await.ok();
        Locator::instance().wallet_controller.delete(config, credentials).await.ok();
    }    
}
```

I guess my point is that if someone wants to do all this, they can. But I don't have a strong opinion on this, so if you'd rather have this implemented and it would aid you, I'll add it.

Reply:
I see, good point. I think we can add some dry run functionality like this later on vcx-napi-rs level.